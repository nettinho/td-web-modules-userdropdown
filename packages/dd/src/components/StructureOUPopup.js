import _ from "lodash/fp";
import React from "react";
import PropTypes from "prop-types";
import { bindActionCreators, compose } from "redux";
import { connect } from "react-redux";
import { Dropdown, Grid, Header, Popup } from "semantic-ui-react";
import { injectIntl } from "react-intl";
import { updateStructure } from "../routines";

export const StructureOUPopup = ({
  id,
  children,
  ou,
  updateStructure,
  domains,
  intl: { formatMessage }
}) => (
  <Popup trigger={children} wide="very" hoverable position="bottom center">
    <Grid centered divided columns={1}>
      <Grid.Row>
        <Grid.Column>
          <Header as="h4">{formatMessage({ id: "structure.ous" })}</Header>
          <Dropdown
            placeholder={formatMessage({ id: "domain.selector.placeholder" })}
            options={domains}
            onChange={(e, { value }) =>
              updateStructure({
                data_structure: { id, ou: value }
              })
            }
            value={ou}
          />
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Popup>
);

StructureOUPopup.propTypes = {
  id: PropTypes.number,
  children: PropTypes.node,
  ou: PropTypes.string,
  updateStructure: PropTypes.func,
  domains: PropTypes.array,
  intl: PropTypes.object
};

const mapStateToProps = ({ structure, domains }) => ({
  domains: domains.map(({ name }, i) => ({ key: i, value: name, text: name })),
  ..._.pick(["id", "ou"])(structure)
});

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ updateStructure }, dispatch)
});

export default compose(
  injectIntl,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(StructureOUPopup);
