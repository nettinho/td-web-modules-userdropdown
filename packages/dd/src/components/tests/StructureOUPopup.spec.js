import React from "react";
import { shallowWithIntl } from "intl-stub";
import { StructureOUPopup } from "../StructureOUPopup";

describe("<StructureOUPopup />", () => {
  it("matches the latest snapshot", () => {
    const props = {
      id: 1,
      children: <h1>Hello</h1>,
      ou: "xxx",
      updateStructure: jest.fn(),
      domains: [{ id: 1, name: "domain 1" }]
    };

    const wrapper = shallowWithIntl(<StructureOUPopup {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
