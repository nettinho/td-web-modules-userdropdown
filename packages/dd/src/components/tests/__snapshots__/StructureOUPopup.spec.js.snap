// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StructureOUPopup /> matches the latest snapshot 1`] = `
<Popup
  hoverable={true}
  keepInViewPort={true}
  on="hover"
  position="bottom center"
  trigger={
    <h1>
      Hello
    </h1>
  }
  wide="very"
>
  <Grid
    centered={true}
    columns={1}
    divided={true}
  >
    <GridRow>
      <GridColumn>
        <Header
          as="h4"
        >
          structure.ous
        </Header>
        <Dropdown
          additionLabel="Add "
          additionPosition="top"
          closeOnBlur={true}
          deburr={false}
          icon="dropdown"
          minCharacters={1}
          noResultsMessage="No results found."
          onChange={[Function]}
          openOnFocus={true}
          options={
            Array [
              Object {
                "id": 1,
                "name": "domain 1",
              },
            ]
          }
          placeholder="domain.selector.placeholder"
          renderLabel={[Function]}
          searchInput="text"
          selectOnBlur={true}
          selectOnNavigation={true}
          value="xxx"
          wrapSelection={true}
        />
      </GridColumn>
    </GridRow>
  </Grid>
</Popup>
`;
